storage/clean/oldDatabaseCleaner.go

```diff
@@ -0,0 +1,242 @@
+package clean
+
+import (
+	"math"
+	"os"
+	"path"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/epochStart/notifier"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/factory/directoryhandler"
+)
+
+var log = logger.GetOrCreate("storage/clean")
+
+// ArgsOldDatabaseCleaner holds the arguments needed for creating an oldDatabaseCleaner
+type ArgsOldDatabaseCleaner struct {
+	DatabasePath        string
+	StorageListProvider StorageListProviderHandler
+	EpochStartNotifier  EpochStartNotifier
+}
+
+type oldDatabaseCleaner struct {
+	sync.RWMutex
+
+	databasePath        string
+	storageListProvider StorageListProviderHandler
+	pathRemover         func(file string) error
+	directoryReader     storage.DirectoryReaderHandler
+	oldestEpochsToKeep  map[uint32]uint32
```

Comment:
why do we need a map here? 
isn't it that we always when going from epoch k-1 to epoch k we remove up to oldestEpochsToKeep from k-1?
then it seems there is no reason to have a map?

Reply:
I know, but a map looks better and I think it's wasier to work with, 
Also, in the future we might want to keep more epochs in this map (for example, delete only the databases marked for deletion 3 epochs ago), so a map would help